testbench/common.py

```diff
@@ -644,27 +647,30 @@ def _extract_data(data):
     return data
 
 
-def __get_streamer_response_fn(database, method, conn, test_id):
+def __get_streamer_response_fn(database, method, conn, test_id, limit=4, chunk_size=4):
     def response_handler(data):
         def streamer():
-            database.dequeue_next_instruction(test_id, method)
             d = _extract_data(data)
-            chunk_size = 4
+            bytes_yield = 0
             for r in range(0, len(d), chunk_size):
-                if r < 10:
+                if bytes_yield < limit:
```

Comment:
I think if you want to preserve existing default behavior, you would have to choose a larger default limit than 4, correct? Maybe it doesn't matter too much.

Reply:
Yep, I considered that but given the updated configurability/logic, I think it's more straightforward to return a broken stream after 4 bytes as default behavior.

Reply:
That's fine, just wanted to point it out.